README
This repository contains the debian directories associated with the various
Debian packages I'm maintaining.


chef/changelog
chef (0.7.10-0ubuntu1) karmic; urgency=low

  * New upstream release
  * Upstream patch for couchdb 0.10.0 (CHEF-515, quilt patch removed)
  * Fix client logging to file (CHEF-287)
  * Fix config log_location settings (CHEF-500)
  * Requires libmixlib-config-ruby (>= 1.0.12)
  * Suggest, not recommend, rubygems (LP: #424131)

 -- Joshua Timberman <joshua@opscode.com>  Fri, 04 Sep 2009 09:31:46 -0600

chef (0.7.8-0ubuntu2) karmic; urgency=low

  * Add couchdb_0.10.0_support.patch patch to make chef-server working with
    the latest couchdb release (LP: #419112).
  * Bump Standards.

 -- Alessio Treglia <quadrispro@ubuntu.com>  Wed, 26 Aug 2009 12:17:26 +0200

chef (0.7.8-0ubuntu1) karmic; urgency=low

   * Initial release (Closes: #529392, LP: #404403)

 -- Joshua Timberman <joshua@opscode.com>  Wed, 12 Aug 2009 12:18:48 -0600


chef/client.rb
# Configuration File For Chef (chef-client)
# 
# The chef-client program will connect the local system to the specified
# server URLs through a RESTful API to retrieve its configuration.
# 
# By default, the client is configured to connect to a Chef Server
# running on the local system. Change this to reflect your environment.
#
# Chef uses a Ruby DSL for configuration, and this file contains a few
# Ruby idioms. First, symbols. These are designated by a colon sigil, ie,
# :value. Second, in Ruby, everything but false and nil (no quotes or other
# designations) is true, including true, the integer 0 and the string "false".
# So to set the value of a setting to false, write:
#
# some_setting false
#
# Third, Ruby class methods can be used, for example we tell the log to show
# the current time stamp with Chef::Log::Formatter.show_time, below.
#
# log_level specifies the level of verbosity for output.
# valid values are: :debug, :info, :warn, :error, :fatal

log_level          :info

# log_location specifies where the client should log to.
# valid values are: a quoted string specifying a file, or STDOUT with
# no quotes. When run as a daemon (default), STDOUT will produce no output.

log_location       "/var/log/chef/client.log"

# ssl_verify_mode specifies if the REST client should verify SSL certificates.
# valid values are :verify_none, :verify_peer. The default Chef Server 
# installation on Debian will use a self-generated SSL certificate so this
# should be :verify_none unless you replace the certificate.

ssl_verify_mode    :verify_none

# Server URLs.
# 
# registration_url specifies the URL which the client retrieves to register.
# valid values are any HTTP URL.

registration_url   "https://localhost"

# openid_url specifies the URL where the server's OpenID relay is listening.
# valid values are any HTTP URL. The default server configuration is set to
# use a vhost running on port 444 for this.
#
# NOTE: The client/server openid communication will be removed in favor of a
# pre-shared key authentication and authorization architecture in a future 
# release of Chef. This is currently scheduled for version 0.8.0.

openid_url         "https://localhost:444"

# template_url specifies the URL where the client should retrieve templates.
# valid values are any HTTP URL.

template_url       "https://localhost"

# remotefile_url specifies the URL where the client should retrieve remote 
# static file and directory contents.
# valid values are any HTTP URL.

remotefile_url     "https://localhost"

# search_url specifies the URL where the client should send queries for search
# indexes.
# valid values are any HTTP URL.

search_url         "https://localhost"

# role_url specifies the URL where the client should look for role data.
# valid values are any HTTP URL.

role_url           "https://localhost"

# file_cache_path specifies where the client should cache cookbooks, server
# cookie ID, and openid registration data.
# valid value is any filesystem directory location.

file_cache_path    "/var/cache/chef"

# pid_file specifies the location of where chef-client daemon should keep the pid
# file.
# valid value is any filesystem file location.

pid_file           "/var/run/chef/client.pid"

# Chef::Log::Formatter.show_time specifies whether the chef-client log should
# contain timestamps. 
# valid values are true or false (no quotes, see above about Ruby idioms). The
# printed timestamp is rfc2822, for example:
# Fri, 31 Jul 2009 19:19:46 -0600

Chef::Log::Formatter.show_time = true



chef/indexer.rb
# Configuration File For Chef Indexer (chef-indexer) 
#
# The chef-indexer program runs on the Chef Server to generate search indexes
# of node data stored in the Server's CouchDB store. Currently, the search 
# uses ferret, a pure-Ruby search library. In a future version of Chef this 
# will change, so please be aware.
#
# Chef uses a Ruby DSL for configuration, and this file may contain some
# Ruby idioms. First, symbols. These are designated by a colon sigil, ie,
# :value. Second, in Ruby, everything but false and nil (no quotes or other
# designations) is true, including true, the integer 0 and the string "false".
# So to set the value of a setting to false, write:
#
# some_setting false
#
# Third, Ruby class methods can be used, for example we tell the log to show
# the current time stamp with Chef::Log::Formatter.show_time, below.
#
# log_location specifies where the indexer should log to.
# valid values are: a quoted string specifying a file, or STDOUT with
# no quotes. When run as a daemon (default), STDOUT will produce no output.

log_location       "/var/log/chef/indexer.log"

# search_index_path specifies where the indexer should store the indexes.
# valid value is any filesystem directory location.

search_index_path    "/var/lib/chef/search_index"

# pid_file specifies the location of where chef-indexer daemon should keep the 
# pid file.
# valid value is any filesystem file location.

pid_file           "/var/run/chef-indexer.pid"

# Chef::Log::Formatter.show_time specifies whether the chef-client log should
# contain timestamps. 
# valid values are true or false (no quotes, see above about Ruby idioms). The
# printed timestamp is rfc2822, for example:
# Fri, 31 Jul 2009 19:19:46 -0600

Chef::Log::Formatter.show_time = true



stompserver/stompserver-0.9.9/debian/manpage.1.ex
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH STOMPSERVER SECTION "July 17, 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
stompserver \- program to do something
.SH SYNOPSIS
.B stompserver
.RI [ options ] " files" ...
.br
.B bar
.RI [ options ] " files" ...
.SH DESCRIPTION
This manual page documents briefly the
.B stompserver
and
.B bar
commands.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.
\fBstompserver\fP is a program that...
.SH OPTIONS
These programs follow the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options is included below.
For a complete description, see the Info files.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.SH SEE ALSO
.BR bar (1),
.BR baz (1).
.br
The programs are documented fully by
.IR "The Rise and Fall of a Fooish Bar" ,
available via the Info system.
.SH AUTHOR
stompserver was written by <upstream author>.
.PP
This manual page was written by Bryan McLellan <btm@loftninjas.org>,
for the Debian project (and may be used by others).


chef/etc/chef/server.rb
# Configuration File For Chef (chef-server)
# 
# chef-server is a Merb application slice. By default it is configured to
# run via Mongrel, the default Merb adapter. This should be run as:
#
#		chef-server -c2 
#
#	Which specifies two workers, so that chef-server has a worker to handle
# OpenID requests. The two processes will be listening on port 4000 and 4001,
# see the URL settings below.
#
# For details on how to set up the chef-server with a web front end proxy 
# (nginx, apache, etc) or as an application under Passenger, see the Chef 
# Wiki, under Additional Resources.
#
# This file configures the behavior of the running server itself.
#
# Chef uses a Ruby DSL for configuration, and this file contains a few
# Ruby idioms. First, symbols. These are designated by a colon sigil, ie,
# :value. Second, in Ruby, everything but false and nil (no quotes or other
# designations) is true, including true, the integer 0 and the string "false".
# So to set the value of a setting to false, write:
#
# some_setting false
#
# Third, Ruby class methods can be used, for example we tell the log to show
# the current time stamp with Chef::Log::Formatter.show_time, below.
#
# log_level specifies the level of verbosity for output.
# valid values are: :debug, :info, :warn, :error, :fatal

log_level          :info

# log_location specifies where the server should log to.
# valid values are: a quoted string specifying a file, or STDOUT with
# no quotes. This is the application log for the Merb workers that get
# spawned.

log_location       "/var/log/chef/server.log"

# ssl_verify_mode specifies if the REST client should verify SSL certificates.
# valid values are :verify_none, :verify_peer. The default Chef Server 
# installation on Debian will use a self-generated SSL certificate so this
# should be :verify_none unless you replace the certificate.

ssl_verify_mode    :verify_none

# The next few settings specify the URL where chef-server should respond
# for various components. 
#
# registration_url specifies the URL which clients retrieve to register.
# valid values are any HTTP URL.

registration_url   "http://localhost:4000"

# openid_url specifies the URL where the server's OpenID relay is listening.
# valid values are any HTTP URL. This should be one port higher than the other
# URLs.
#
# NOTE: The client/server openid communication will be removed in favor of a
# pre-shared key authentication and authorization architecture in a future 
# release of Chef. This is currently scheduled for version 0.8.0.

openid_url         "http://localhost:4001"

# template_url specifies the URL where clients should retrieve templates.
# valid values are any HTTP URL.

template_url       "http://localhost:4000"

# remotefile_url specifies the URL where clients should retrieve remote 
# static file and directory contents.
# valid values are any HTTP URL.

remotefile_url     "http://localhost:4000"

# search_url specifies the URL where the client should send queries for search
# indexes.
# valid values are any HTTP URL.

search_url         "http://localhost:4000"

# role_url specifies the URL where the client should look for role data.
# valid values are any HTTP URL.

role_url           "http://localhost:4000"

# file_cache_path specifies where the client should cache cookbooks, server
# cookie ID, and openid registration data.
# valid value is any filesystem directory location.

file_cache_path    "/var/cache/chef"

# validation_token specifies a token that clients can use to automatically authorize
# with the server, normally this must be done through the webui.
# valid value is a string. This should be alphanumeric, otherwise unexpected things 
# may occur. This is commented out so the default value gets set to nil (invalid,
# unusable) so we're not shipping a default known value. If you want to use this
# feature, you'll need to uncomment it, set a good value (random strings of 20+ 
# characters work well), and restart the chef-server.

#validation_token   ""

# cookbook_path is a Ruby array of filesystem locations to search for cookbooks.
# valid value is a string, or an array of strings of filesystem directory locations.
# This setting is searched beginning (index 0) to end in order. You might specify
# multiple search paths for cookbooks if you want to use an upstream source, and
# provide localised "site" overrides.
# The default value, /srv/chef/cookbooks does not contain any cookbooks by default.
# See the Chef Wiki for more information about setting up a local repository for
# working on cookbooks.
# http://wiki.opscode.com/display/chef/Chef+Repository

#cookbook_path      [ "/srv/chef/site-cookbooks", "/srv/chef/cookbooks" ]
cookbook_path      [ "/srv/chef/cookbooks" ]

# file_cache_path specifies where the client should cache cookbooks, server
# cookie ID, and openid registration data.
# valid value is any filesystem directory location.

file_cache_path    "/var/cache/chef"

# node_path specifies a location for where to find node-specific recipes.
# valid values are any filesystem direcory location.

node_path          "/srv/chef/nodes"

# openid_store_path specifies a location where to keep openid nonces for clients.
# valid values are any filesystem directory location.

openid_store_path  "/var/lib/chef/openid/store"

# openid_store_path specifies a location where to keep openid nonces for clients.
# valid values are any filesystem directory location.

openid_cstore_path "/var/lib/chef/openid/cstore"

# search_index_path tells the indexer where to store search indexes.
# valid values are any filesystem directory location.

search_index_path  "/var/lib/chef/search_index"

# role_path designates where the server should load role JSON and Ruby DSL 
# files from.
# valid values are any filesystem directory location. Roles are a feature 
# that allow you to easily reuse lists of recipes and attribute settings.
# Please see the Chef Wiki page for information on how to utilize the feature.
# http://wiki.opscode.com/display/chef/Roles

role_path          "/srv/chef/roles"

# Chef::Log::Formatter.show_time specifies whether the chef-client log should
# contain timestamps. 
# valid values are true or false (no quotes, see above about Ruby idioms). The
# printed timestamp is rfc2822, for example:
# Fri, 31 Jul 2009 19:19:46 -0600

Chef::Log::Formatter.show_time = true

